_includes/css/activation.scss.liquid

```diff
@@ -197,25 +249,25 @@
   and children of the current page.
 {%- endcomment %}
 
-{% if activation_third_level_index -%}
+{% if nav_levels[3] -%}
 
 {{ activation_collection_prefix }} > li > a,
 {{ activation_collection_prefix }} > li > ul > li > a,
-{{ activation_collection_prefix }} > li > ul > li > ul > li:not(:nth-child({{ activation_third_level_index }})) > a {
+{{ activation_collection_prefix }} > li > ul > li > ul > li:not(:nth-child({{ nav_levels[3] }})) > a {
```

Comment:
@mattxwang thanks for finding time for re-reviewing this PR – especially as it's probably even harder to review than it was to implement…

In fact the varying sort order of the `0` and `0.0` pages was already [mentioned](https://github.com/just-the-docs/just-the-docs/pull/1244#issuecomment-1656819401) when testing #1244, which was merged in v0.6.0. So it's definitely not caused by the present PR.

Try running the following code:

```liquid
{% assign a = "" | split: "" | push: 0 | push: 0.0 | sort %}
{% assign b = "" | split: "" | push: 0.0 | push: 0 | sort %}

- {{ a | inspect }}
- {{ b | inspect }}
```

Testing locally, that produces:

![image](https://github.com/just-the-docs/just-the-docs/assets/18308236/6ff1a9af-fded-4c5d-bded-18f1f289901a)

I suspect that the order of the arrays of pages accessed via `site` variables is (at least to some extent) random. Since sorting doesn't determine a specific order for `0` and `0.0`, the nav links to pages with those values can appear in either order.

I should probably remove one of the pages, and add a warning about mixing integers and floats. Then the false alarm in the diff will definitely disappear! 😃

Reply:
Oh! Wow, that completely slipped my mind - it's been a while since I reviewed #1244. Thank you for the patient explanation! Apologies for leading you down an incorrect rabbit hole re: `nav_levels`.

> I suspect that the order of the arrays of pages accessed via site variables is (at least to some extent) random. Since sorting doesn't determine a specific order for 0 and 0.0, the nav links to pages with those values can appear in either order.

Shopify's [`sort`](https://shopify.github.io/liquid/filters/sort/) documentation doesn't mention anything about stable sorting, but my (uninformed) guess is that `0` and `0.0` are both just [`Numbers`](https://shopify.github.io/liquid/basics/types/#number) and must evaluate to the same underlying number (which might cause this behaviour - your example is very helpful here). I'm realizing now that I have no concrete idea of how Liquid's type/value system really works, when I probably should!

About to hop into a few meetings, but I'll take one last look and likely approve; the comments you added in `activation.scss.liquid` (w/ the CFG) were quite helpful. I might also look through the Liquid source code and see if I can get any hints as to why this behaviour might happen, but that shouldn't block this PR.